Content.Server/Abilities/Goliath/GoliathTentacleSystem.cs

```diff
@@ -0,0 +1,72 @@
+using Content.Server.NPC.HTN;
+using Content.Shared.Abilities.Goliath;
+using Content.Shared.Abilities.Goliath.Components;
+using Content.Shared.Actions;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Abilities.Goliath;
+
+public sealed class GoliathTentacleSystem : SharedGoliathTentacleSystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly SharedActionsSystem _actions = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GoliathTentacleUserComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(Entity<GoliathTentacleUserComponent> ent, ref MapInitEvent args)
+    {
+        ent.Comp.TentacleAction = _actions.AddAction(ent, ent.Comp.TentacleActionId);
+    }
+
```

Comment:
This is also stinky but I can probably repurpose it into NPC stuff so I'm fine with refactoring this later.